src/smokeTest/java/uk/gov/hmcts/reform/draftstore/client/IdamClient.java

```diff
@@ -0,0 +1,150 @@
+            .body(buildUserRegistrationRequestBody())
+            .post(idamUrl + "/testing-support/accounts")
+            .then()
+            .statusCode(NO_CONTENT.value());
+    }
+
+    public Optional<String> getAuthorisationCode(
+        boolean failIfUnauthorised
+    ) throws UnsupportedEncodingException {
+
+        Response response = sendAuthorisationRequest();
+
+        if (response.getStatusCode() == OK.value()) {
+            return Optional.of(extractAuthorisationCodeFromIdamResponse(response));
+        } else if (response.getStatusCode() == UNAUTHORIZED.value() && !failIfUnauthorised) {
+            return Optional.empty();
+        } else {
+            String msgFormat =
+                "Unexpected Idam response (%s) when trying to log user in. Response body: %s";
+
+            throw new AssertionError(String.format(
+                msgFormat,
+                response.getStatusCode(),
+                response.getBody().print()
+            ));
+        }
+    }
+
+    public String getIdamToken(String code) {
+        return RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(this.idamUrl)
+            .header(CONTENT_TYPE, APPLICATION_FORM_URLENCODED_VALUE)
+            .queryParams(
+                ImmutableMap.of(
+                    "grant_type", "authorization_code",
+                    "client_id", clientId,
+                    "client_secret", clientSecret,
+                    "redirect_uri", redirectUri,
+                    "code", code
+                )
+            )
+            .post("/oauth2/token")
+            .then()
+            .statusCode(200)
+            .extract()
+            .body()
+            .jsonPath()
+            .get("access_token");
+    }
+
+    private Response sendAuthorisationRequest() throws UnsupportedEncodingException {
+        return RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(this.idamUrl)
+            .header("Authorization", "Basic " + buildIdamSignInToken())
+            .queryParams(
+                ImmutableMap.of(
+                    "response_type", "code",
+                    "client_id", clientId,
+                    "client_secret", clientSecret,
+                    "redirect_uri", redirectUri
+                )
+            )
+            .post("/oauth2/authorize")
+            .thenReturn();
+    }
+
+    private String buildIdamSignInToken() throws UnsupportedEncodingException {
+        String unencodedToken = String.format("%s:%s", email, password);
+
+        return Base64.getEncoder().encodeToString(
+            unencodedToken.getBytes("UTF-8")
+        );
+    }
+
+    private Map<String, String> buildUserRegistrationRequestBody() {
+        return ImmutableMap.of(
+            "email", this.email,
+            "forename", "Platform Engineering",
+            "surname", "Smoke tests",
+            "password", this.password
+        );
+    }
```

Comment:
used once, imho worth inlining. similar to what do did in `sendAuthorisationRequest()`

Reply:
Made the metod a bit too long and reduced its readability. That's why I extracted it.